lib/engine/game/g_18_india/step/track.rb

```diff
@@ -1,15 +1,108 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/track'
+require_relative 'tracker'
 
 module Engine
   module Game
     module G18India
       module Step
         class Track < Engine::Step::Track
+          include Tracker
+
+          # modified to add remove gauge change option in phase IV
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] unless @round.removed_gauge.empty?
+            return [] if entity.company? || !can_lay_tile?(entity)
+
+            actions = %w[lay_tile pass]
+            actions << 'remove_gauge_change' if may_remove_gauge_change?(entity)
+            actions
+          end
+
+          def setup
+            super
+            @round.removed_gauge = []
+          end
+
+          def round_state
+            super.merge({ removed_gauge: [] })
+          end
+
+          def description
+            tile_lay = get_tile_lay(current_entity)
+            return 'Lay Track' unless tile_lay
+
+            if tile_lay[:lay] && tile_lay[:upgrade]
+              @game.phase.name == 'IV' ? 'Lay/Upgrade Track OR Remove Gauge Change Marker' : 'Lay/Upgrade Track'
+            elsif tile_lay[:lay]
+              'Lay Track'
+            else
+              'Upgrade Track'
+            end
+          end
+
+          # ------ Show a note that gauge changes may be removed ------
+
+          def note_text_block
+            [
+              'May remove a Gauge Change Marker as a track action.',
+              'Click on a connected marker to remove it.',
+            ]
+          end
+
+          def show_note?(entity)
+            may_remove_gauge_change?(entity)
+          end
+
+          # ------ Code for 'remove_gauge_change' Action [Remove Gauge Change Markers in Phase IV] ------
+
+          def may_remove_gauge_change?(entity)
+            num_track_actions = @round.num_laid_track + @round.num_upgraded_track
+            @game.phase.name == 'IV' && num_track_actions.zero? && any_gauge_changes? && any_reachable_gauge_changes?(entity)
```

Comment:
Thanks for this note.  I learned a bit about this as I was working on walking a Track Laying route from a railhead.  

I agree with your suggestion to remove the `any_reachable_gauge_changes?(entity)` check from the method, as that is checked in the process call as you noted. 

The only downside is you may be shown the message that Gauge Change Markers may be removed, because they are on the map, but can't actually remove it as those markers are out of reach for the current corp. 

Thanks!

Reply:
Another idea would be to add a `return if @game.loading` guard clause in the `reachable_gauge_change?` method?

Reply:
Since `reachable_gauge_change?` is called from `process_remove_gauge_change`, which will also be called when running though all the actions on game load, I think the `game.loading` check will be more effective. 

Added the guard clause.

Reply:
No, that's the wrong way to do this.
The issue that you've now got is that `actions` will return different lists of possible actions when running interactively or non-interactively (game loading). That's potentially going to cause trouble.
I suspect that it probably wouldn't actually be an issue here, but it's something to be avoided. The problem could occur if `actions` returns an empty array when called from the game UI, then that player's turn would be skipped. If on game loading `actions` returns a non-empty array then their turn will not be skipped, and the game will break.
You want to move the `@game.loading` check to inside `process_remove_gauge_change`, so that the call to `reachable_gauge_change?` isn't made if `@game.loading` is true. That way the legality of the action is checked when it is first made, then when loading the game again we assume that the move is legal as it's been checked before getting added to the game history.

Reply:
I understand, thank you for the explanation.  I will adjust.

Reply:
Moved loading check.

Reply:
> The only downside is you may be shown the message that Gauge Change Markers may be removed, because they are on the map, but can't actually remove it as those markers are out of reach for the current corp.

You can keep a graph check for reachable gauge change markers in `show_note?` as that method won't be called during the game load process.